------------------------------------------------------------
revno: 14125
revision-id: squid3@treenet.co.nz-20161215093634-ykbs6tv8pdusz7cj
parent: squid3@treenet.co.nz-20161215092210-8gupdsihb4d8fufk
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=3940
author: Garri Djavadyan <garryd@comnet.uz>
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Thu 2016-12-15 22:36:34 +1300
message:
  Bug 3940 (partial): hostHeaderVerify failures MISS when they should be HIT
  
  This fixes the critical condition leading to the HIT. However not all
  code is correctly setting flags.noCache and flags.cacheable (see bugzilla).
  So there may be other fixes needed after this.
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20161215093634-ykbs6tv8pdusz7cj
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: 3e1ebda070635dcabfa4f77d697ac12e8683106f
# timestamp: 2016-12-15 09:39:01 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20161215092210-\
#   8gupdsihb4d8fufk
# 
# Begin patch
=== modified file 'src/client_side_reply.cc'
--- src/client_side_reply.cc	2016-11-11 06:03:25 +0000
+++ src/client_side_reply.cc	2016-12-15 09:36:34 +0000
@@ -1649,7 +1649,9 @@
 {
     HttpRequest *r = http->request;
 
-    if (r->flags.cachable || r->flags.internal) {
+    // client sent CC:no-cache or some other condition has been
+    // encountered which prevents delivering a public/cached object.
+    if (!r->flags.noCache || r->flags.internal) {
         lookingforstore = 5;
         StoreEntry::getPublicByRequest (this, r);
     } else {

